toolchains/clang.cmake

```diff
@@ -12,9 +12,16 @@ include( "${CMAKE_CURRENT_LIST_DIR}/gcc_compatibles.cmake" )
 
 list( APPEND TNUN_compiler_optimize_for_speed -fvectorize -fslp-vectorize -fslp-vectorize-aggressive )
 list( APPEND TNUN_compiler_report_optimization -Rpass=loop-.* )
+list( APPEND TNUN_disabled_warnings -Wno-unknown-warning-option "-Wno-error=#warnings" -Wno-attributes $<$<COMPILE_LANGUAGE:CXX>:-Wno-overloaded-virtual> )
 
 # http://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
 # http://clang.llvm.org/docs/UsersManual.html#controlling-code-generation
-set( TNUN_compiler_runtime_sanity_checks -fsanitize=undefined -fsanitize=integer )
+set( TNUN_compiler_runtime_sanity_checks -fsanitize=undefined -fsanitize=integer -fsanitize=address -fno-omit-frame-pointer )
 
-add_compile_options( -Wheader-guard )
+if( "${CMAKE_CXX_COMPILER}" MATCHES ".*ccache" )
```

Comment:
When clang is used behind ccache, it throws a lot of "unused-argument" warnings. I'm not sure why that happens (it also happens on clang for Android, both from ndk-build and cmake android build). Having this much warnings for each and every source file is very annoying.